pyscf/adc/radc.py

```diff
@@ -97,42 +100,61 @@ def compute_amplitudes(myadc, eris):
     eris_ovov = eris.ovov
     eris_oovv = eris.oovv
     eris_ovvo = eris.ovvo
-    eris_ovvv = eris.ovvv
 
-    e = myadc.mo_energy
+    # Compute first-order doubles t2 (tijab)
 
+    v2e_oovv = eris_ovov[:].transpose(0,2,1,3).copy()
+    e = myadc.mo_energy
     d_ij = e[:nocc][:,None] + e[:nocc]
-
     d_ab = e[nocc:][:,None] + e[nocc:]
 
     D2 = d_ij.reshape(-1,1) - d_ab.reshape(-1)
-
     D2 = D2.reshape((nocc,nocc,nvir,nvir))
 
     D1 = e[:nocc][:None].reshape(-1,1) - e[nocc:].reshape(-1)
     D1 = D1.reshape((nocc,nvir))
 
-    # Compute first-order doubles t2 (tijab)
-
-    v2e_oovv = eris_ovov[:].transpose(0,2,1,3).copy()
-
     t2_1 = v2e_oovv/D2
+    del (v2e_oovv)
+    del (D2)
+
+    cput0 = log.timer_debug1("Completed t2_1 amplitude calculation", *cput0)
 
     # Compute second-order singles t1 (tij)
 
-    t2_1_a = t2_1 - t2_1.transpose(1,0,2,3).copy()
+    if isinstance(eris.ovvv, type(None)):
+        chnk_size = radc_ao2mo.calculate_chunk_size(myadc)
+    else :
+        chnk_size = nocc
+    a = 0
+    t1_2 = np.zeros((nocc,nvir))
 
-    eris_ovvv = radc_ao2mo.unpack_eri_1(eris.ovvv, nvir)
-    t1_2 = 0.5*lib.einsum('kdac,ikcd->ia',eris_ovvv,t2_1_a,optimize=True)
-    t1_2 -= 0.5*lib.einsum('kcad,ikcd->ia',eris_ovvv,t2_1_a,optimize=True)
-    t1_2 += lib.einsum('kdac,ikcd->ia',eris_ovvv,t2_1,optimize=True)
-    del eris_ovvv
-    t1_2 -= 0.5*lib.einsum('lcki,klac->ia',eris_ovoo,t2_1_a,optimize=True)
-    t1_2 -= 0.5*lib.einsum('kcli,lkac->ia',eris_ovoo,t2_1_a,optimize=True)
+    for p in range(0,nocc,chnk_size):
+        if getattr(myadc, 'with_df', None):
+            eris_ovvv = dfadc.get_ovvv_df(myadc, eris.Lov, eris.Lvv, p, chnk_size).reshape(-1,nvir,nvir,nvir)
+        else :
+            eris_ovvv = radc_ao2mo.unpack_eri_1(eris.ovvv, nvir)
```

Comment:
Does line 136 correspond to in-core algorithm? If so, where is the out-of-core algorithm implementation?

Reply:
Line 136 corresponds to both in-core and out-of-core algorithm. ovvv is packed in the out-of-core algorithm as well